### PR TITLE
linuxPackages.cryptodev: 1.13 -> 1.14

### PR DESCRIPTION
**Pull request by blitz**
## Things done

Update cryptodev to latest version to fix build failure for Linux 5.10.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by blitz** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 359871 --checkout commit`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_4_hardened.cryptodev</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_latest_libre.cryptodev</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 17 packages built:</summary>
  <ul>
    <li>cryptodev (linuxKernel.packages.linux_6_6.cryptodev)</li>
    <li>linuxKernel.packages.linux_5_10.cryptodev</li>
    <li>linuxKernel.packages.linux_5_10_hardened.cryptodev</li>
    <li>linuxKernel.packages.linux_5_15.cryptodev</li>
    <li>linuxKernel.packages.linux_5_15_hardened.cryptodev</li>
    <li>linuxKernel.packages.linux_5_4.cryptodev</li>
    <li>linuxKernel.packages.linux_6_1.cryptodev</li>
    <li>linuxKernel.packages.linux_6_11.cryptodev</li>
    <li>linuxKernel.packages.linux_6_11_hardened.cryptodev</li>
    <li>linuxKernel.packages.linux_6_12.cryptodev</li>
    <li>linuxKernel.packages.linux_6_1_hardened.cryptodev</li>
    <li>linuxKernel.packages.linux_hardened.cryptodev (linuxKernel.packages.linux_6_6_hardened.cryptodev)</li>
    <li>linuxKernel.packages.linux_libre.cryptodev</li>
    <li>linuxKernel.packages.linux_lqx.cryptodev</li>
    <li>linuxKernel.packages.linux_xanmod.cryptodev</li>
    <li>linuxKernel.packages.linux_xanmod_latest.cryptodev (linuxKernel.packages.linux_xanmod_stable.cryptodev)</li>
    <li>linuxKernel.packages.linux_zen.cryptodev</li>
  </ul>
</details>

**Review by MakiseKurisu — APPROVED** [review]

**Review by moni-dz — APPROVED** [review]

File: pkgs/os-specific/linux/cryptodev/default.nix [hunk 1]
```diff
@@ -2,28 +2,20 @@
...
   fetchFromGitHub,
-  fetchpatch,
   kernel ? false,
 }:
 
-stdenv.mkDerivation rec {
-  pname = "cryptodev-linux-1.13";
-  name = "${pname}-${kernel.version}";
+stdenv.mkDerivation (finalAttrs: {
+  pname = "cryptodev-linux-1.14";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/cryptodev/default.nix:9]:
```suggestion
  pname = "cryptodev-linux";
  version = "1.14";
```

File: pkgs/os-specific/linux/cryptodev/default.nix [hunk 2]
```diff
@@ -2,28 +2,20 @@
...
+  name = "${finalAttrs.pname}-${kernel.version}";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/cryptodev/default.nix:10]:
```suggestion
  name = "${finalAttrs.pname}-${kernel.version}-${finalAttrs.version}";
```

File: pkgs/os-specific/linux/cryptodev/default.nix [hunk 3]
```diff
@@ -2,28 +2,20 @@
...
 
   src = fetchFromGitHub {
     owner = "cryptodev-linux";
     repo = "cryptodev-linux";
-    rev = pname;
-    hash = "sha256-EzTPoKYa+XWOAa/Dk7ru02JmlymHeXVX7RMmEoJ1OT0=";
+    rev = finalAttrs.pname;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/cryptodev/default.nix:15]:
```suggestion
    rev = finalAttrs.version;
```

File: pkgs/os-specific/linux/cryptodev/default.nix (hunk 1 above)
Comment (blitz) [reply pkgs/os-specific/linux/cryptodev/default.nix]: @SuperSandro2000 Thanks for the review! I've removed the code that appended the kernel version to the derivation name and applied the remaining fixes you suggested. Now it mostly does what other out-of-tree modules do as well.

**Comment by blitz** [comment]
Result of `nixpkgs-review pr 359871 --checkout commit` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_4_hardened.cryptodev</li>
  </ul>
</details>
<details>
  <summary>18 packages built:</summary>
  <ul>
    <li>cryptodev</li>
    <li>linuxKernel.packages.linux_5_10.cryptodev</li>
    <li>linuxKernel.packages.linux_5_10_hardened.cryptodev</li>
    <li>linuxKernel.packages.linux_5_15.cryptodev</li>
    <li>linuxKernel.packages.linux_5_15_hardened.cryptodev</li>
    <li>linuxKernel.packages.linux_5_4.cryptodev</li>
    <li>linuxKernel.packages.linux_6_1.cryptodev</li>
    <li>linuxKernel.packages.linux_6_11.cryptodev</li>
    <li>linuxKernel.packages.linux_6_11_hardened.cryptodev</li>
    <li>linuxKernel.packages.linux_6_12.cryptodev</li>
    <li>linuxKernel.packages.linux_6_1_hardened.cryptodev</li>
    <li>linuxKernel.packages.linux_hardened.cryptodev (linuxKernel.packages.linux_6_6_hardened.cryptodev)</li>
    <li>linuxKernel.packages.linux_latest_libre.cryptodev</li>
    <li>linuxKernel.packages.linux_libre.cryptodev</li>
    <li>linuxKernel.packages.linux_lqx.cryptodev</li>
    <li>linuxKernel.packages.linux_xanmod.cryptodev</li>
    <li>linuxKernel.packages.linux_xanmod_latest.cryptodev (linuxKernel.packages.linux_xanmod_stable.cryptodev)</li>
    <li>linuxKernel.packages.linux_zen.cryptodev</li>
  </ul>
</details>

**Pull request merged**